src/components/flowchart/draw.js

```diff
@@ -343,11 +367,33 @@ export const drawEdges = function (changed) {
     this.el.edges = this.el.edgeGroup.selectAll('.pipeline-edge');
   }
 
-  if (changed('edges', 'clickedNode', 'linkedNodes')) {
+  if (
+    changed(
+      'edges',
+      'clickedNode',
+      'linkedNodes',
+      'newParamsFlag',
+      'focusMode',
+      'inputOutputDataEdges'
+    )
+  ) {
     allEdges
       .classed(
         'pipeline-edge--parameters',
-        (edge) => edge.sourceNode.type === 'parameters'
+        (edge) =>
+          edge.sourceNode.type === 'parameters' &&
+          !inputOutputDataEdges[edge.id]
```

Comment:
Sorry I don't understand why this condition `!inputOutputDataEdges[edge.id]` is needed here. Is it to fix the problem you identified the other day?

Reply:
this condition is needed to prevent applying this class to parameter nodes that are also input/output nodes - if this check is not applied this will create potential conflicts with the css class that will be applied to parameter nodes that are part of input/output. ( i.e taking this check out will result in some input/output parameter nodes not adopting the `pipeline-edge--parameters--input` class. )